File: server/FunctionalTests/Tests/Rojo04/GitAdd.py
"""
TheNexusAvenger

Tests that the git add request acts correctly.
"""

import NexusGitFunctionalTest



"""
Tests the git add as a standalone request.
"""
class Rojo04TestGitAddAsStandaloneRequest(NexusGitFunctionalTest.NexusGitFunctionalTest):
    """
    Setup for the test.
    """
    def setupTest(self):
        self.workspace.writeFile("rojo.json","{\"name\": \"Nexus Git Repository\",\"servePort\": 20001,\"partitions\": {\"src\": {\"path\": \"src\",\"target\": \"ReplicatedStorage.NexusGit\"},\"test\": {\"path\": \"test\",\"target\":\"ReplicatedStorage.NexusGitTest\"}}}")
        self.workspace.createDirectory("src")
        self.workspace.createDirectory("test")
        self.workspace.writeFile("src/testscript1.server.lua","print(\"Hello world 1!\")")
        self.workspace.writeFile("src/testscript2.client.lua","print(\"Hello world 2!\")")
        self.workspace.writeFile("test/testscript3.lua","print(\"Hello world 3!\")")

        # Initialize the git repository.
        self.workspace.runCommand("git init")

    """
    Runs the test.
    """
    def runTest(self):
        # Wait for it to initialize.
        self.setPortNumber(20001)
        self.waitForInitialization()

        # Send a standalone request and get the response.
        response = self.sendPOSTRequest("/gitadd","[\"src/testscript1.server.lua\",\"test/testscript3.lua\"]")
        self.assertEquals(response,"[\"Add complete.\"]")

        # Assert the status is correct.
        response = self.sendGETRequest("/gitstatus")
        self.assertEquals(response,"[\"Current branch:\",\"master\",\"Changes to be committed:\",\"New file: src/testscript1.server.lua\",\"New file: test/testscript3.lua\",\"Untracked files:\",\"rojo.json\",\"src/testscript2.client.lua\"]")

File: server/NexusGitTests/Functional/Rojo04/GitAdd.cs
/*
 * TheNexusAvenger
 * 
 * Tests the GitAdd request with Rojo 0.4.
 */

using NUnit.Framework;

namespace NexusGitTests.Functional.Rojo04
{
    /*
     * Tests adding as a standalone request.
     */
    public class TestGitAdd : BaseFunctionalTest
    {
        /*
         * Sets up the test.
         */
        public override void Setup()
        {
            // Write the files.
            this.Workspace.WriteFile("rojo.json","{\"name\": \"Nexus Git Repository\",\"servePort\": 20001,\"partitions\": {\"src\": {\"path\": \"src\",\"target\": \"ReplicatedStorage.NexusGit\"},\"test\": {\"path\": \"test\",\"target\":\"ReplicatedStorage.NexusGitTest\"}}}");
            this.Workspace.WriteFile("file1.txt","Test file 1");
            this.Workspace.WriteFile("file2.txt","Test file 2");
            this.Workspace.WriteFile("file3.txt","Test file 3");
            
            // Initialize the repository.
            this.Workspace.RunCommand("git","init");
        }
        
        /*
         * Runs the test.
         */
        public override void Run()
        {
            // Add the file and assert the response is correct.
            var response = this.SendPOSTRequest("/gitadd", "[\"file1.txt\",\"file3.txt\"]");
            Assert.AreEqual(response,"[\"Add complete.\"]");
            
            // Assert the initial status is correct.
            response = this.SendGETRequest("/getgitstatus");
            Assert.AreEqual(response,"[\"Current branch:\",\"master\",\"Changes to be committed:\",\"New file: file1.txt\",\"New file: file3.txt\",\"Untracked files:\",\"file2.txt\",\"rojo.json\"]");
        }
    }
}